src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,123 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import core.basesyntax.service.InvalidDataException;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static RegistrationServiceImpl registrationService;
+    private User testUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        testUser = new User();
+    }
+
+    @Test
+    void register_newUser_Ok() {
+        testUser.setLogin("validTestUser");
+        testUser.setAge(50);
+        testUser.setPassword("incred1blePassw0rd");
+        assertEquals(testUser, registrationService.register(testUser));
+    }
+
+    @Test
+    void register_userExists_notOk() {
+        testUser.setLogin("unitTestUserLogin");
+        Storage.people.add(testUser);
```

Comment:
Ensure that the `Storage.people` list is cleared before each test to avoid interference between tests. This can be done in the `setUp` method.